tests/temp/tempFileTestInternalApi.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+require_once(dirname(__FILE__) . "/../../interface/globals.php");
+
+
+// Looking into how to use the api locally via php calls (4 methods are shown below)
+
+
+// GUZZLE CALL OF API
+//  noted needed fix for when using localhost with a port when using the server
+//  does make me wonder if needing to use the server name for local api's will be very tough, sometimes impossible, to support, though
+//   (for example, if trying to run this through a php-fpm instance that does not include apache/nginx then not sure how it would work)
+use OpenEMR\Common\Http\oeHttp;
+
+error_log("DEBUG1: " . $_SERVER['HTTP_HOST'] . " : " . $GLOBALS['web_root']);
+$server = (preg_match('/^localhost:[0-9]+$/', $_SERVER['HTTP_HOST'])) ? 'localhost' : $_SERVER['HTTP_HOST'];
+$response = oeHttp::setLocalApiContext(true)->get($server . $GLOBALS['web_root'] . '/apis/api/facility');
+echo $response->body();
+
+
+// BYPASS GUZZLE AND DIRECTLY CALL API via route handler
+//  Needed to comment out pertinent authorization_check call for this to work
+use OpenEMR\Common\Http\HttpRestRouteHandler;
+
+require_once(dirname(__FILE__) . "/../../_rest_config.php");
+$gbl = RestConfig::GetInstance();
+HttpRestRouteHandler::dispatch($gbl::$ROUTE_MAP, '/api/facility', "GET");
+
+
+// BYPASS GUZZLE AND USE THE SERVICE WITHOUT REST CONTROLLER
+use OpenEMR\Services\FacilityService;
+
+echo json_encode((new FacilityService())->getAll());
```

Comment:
Guessing this facility service may be the way to go, but will await @sjpadgett input and @adunsulag input on this.

Reply:
hmm, but would be a heck of a lot of work to replace the following with services:
https://github.com/openemr/openemr/blob/master/phpfhir/libs/controller/ClientAppController.php#L73
(what to do?)

Reply:
you're talking about how we are just forwarding all calls from the FHIR api to the equivalent non FHIR api? I'm pretty sure @sjpadgett Wrote that code as an example, I don't think we would want much of the codebase to do what he is doing in that specific instance.

I think to provide clarity, just think about it if its written in PHP and its inside the OpenEMR directory we want people working with objects and the services (ie the Application Layer of the system).

If it's javascript that is inside of OpenEMR or if it's an application that runs outside of the OpenEMR codebase it will use the REST apis.

That's my 2 cents in there but @sjpadgett might have more to add.

Reply:
Originally this rest call was an instantiated controller which was a good place to demo internal rest calls. Chances are at some point when/if we get serious about moving to our controllers/services, these controller base classes are going to be extended or abstracted in one way or another to meet business logic so, consistency at the application layer is important. 

I think internal calls have a place for cases like modules that act as internal clients to OpenEMR. CCM is an example where many UI actions are dispatched calls. Though I don't take exception with using REST calls in the Fhir Provider client, in the end, it simply will not be practical in some cases. However, many external Fhir server iterops will have openemr acting as client so, I don't see any reasons Provider Client can't act as a client to openemr by using the same benefits of Rest api that an external Fhir client to openemr enjoys. Where possible of course.

Originally I had dispatch allowing a hook via a parameter that returned the controller response objects back to call, in effect, bypassing dispatching to Http and thus, rest controller. This gave us the benefit of using the rest api for internal calls without external exposure. At the time I thought this was clever but nobody responded to the proposal so, I left it out(I remember thinking to myself at the time that it must be a really dumb idea for no one to offer up opinions).

So bottom line Brady, i'd not worry too much with the Provider Client for now because it will change drastically here soon from a test jig to an actual useful client. Also note(in case nobody has paid attention to it) I have a dispatch class I wrote that may or may not prove useful as dispatcher for UI layer actions. I use all the time for other apps i've developed for openemr add ons and makes interactions with business layer easier. Just know it is there but I don't know if useful in grand scheme of things. phpfhir/libs/controller/oeDispatcher.php.